src/main/java/com/fcgl/madrid/search/service/UserSearchService.java

```diff
@@ -1,18 +1,69 @@
 package com.fcgl.madrid.search.service;
 
-import com.fcgl.madrid.search.dataModel.Category;
+import com.fcgl.madrid.search.dataModel.UserSearch;
 import com.fcgl.madrid.search.payload.InternalStatus;
+import com.fcgl.madrid.search.payload.request.UserSearchRequest;
 import com.fcgl.madrid.search.payload.response.Response;
-import com.fcgl.madrid.search.repository.CategoryRepository;
+import com.fcgl.madrid.search.repository.UserSearchRepository;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.SpringApplication;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
 
 @Service
 public class UserSearchService {
     //TODO: Add necessary functions
+
+    private UserSearchRepository repo;
+
+    @Autowired
+    public UserSearchService(UserSearchRepository repo) {
+        this.repo = repo;
+    }
+
+    public ResponseEntity<Response<String>> addQueryToTable(UserSearchRequest request) {
+        Long userId = request.getUserId();
+        String query = request.getQuery();
+        Date date = new Date();
+
+        UserSearch model = new UserSearch(userId, query, date);
+        UserSearch resp = repo.save(model); // what does this return if it is successful??
```

Comment:
Is this comment necessary? Does it return the UserSearch object?

Reply:
It maybe throws an exception if there is an error? I'm not sure, would be good to look up what `save` can return or throw.

Any network connection exceptions will be caught by recilience4j when implemented

Reply:
Straggling comment for myself lol. Pretty sure it just returns null.